src/models/inbound/dfspOTPValidate.model.ts

```diff
@@ -0,0 +1,347 @@
+    } catch (error) {
+      const mojaloopError = this.reformatError(error)
+
+      this.logger.push({ error }).error('OTPIsValid -> scopesReceived')
+      this.logger.push({ mojaloopError }).info(`Sending error response to ${toParticipantId}`)
+
+      // note: this is an error code scenario that will need to be addressed.
+      //       something along the lines 6xxx "No scopes granted"
+      await this.thirdpartyRequests.putConsentRequestsError(
+        consentRequestsRequestId,
+        mojaloopError as unknown as fspiopAPI.Schemas.ErrorInformationObject,
+        toParticipantId
+      )
+      // throw error to stop state machine
+      throw error
+    }
+  }
+
+  async onRegisterConsent (): Promise<void> {
+    const { consentRequestsRequestId, toParticipantId, scopes } = this.data
+
+    const postConsentRequestsPayload: tpAPI.Schemas.ConsentsPostRequest = {
+      consentId: uuid(),
+      consentRequestId: consentRequestsRequestId,
+      scopes: scopes?.scopes || []
+    }
+
+    try {
+      await this.thirdpartyRequests.postConsents(
+        postConsentRequestsPayload,
+        toParticipantId
+      )
+    } catch (error) {
+      const mojaloopError = this.reformatError(error)
+
+      this.logger.push({ error }).error('scopesReceived -> consentSent')
+      this.logger.push(error).error('ThirdpartyRequests.postConsents request error')
+
+      // note: if the POST /consents fails at the DFSP we report that back to the pisp
+      await this.thirdpartyRequests.putConsentRequestsError(
+        consentRequestsRequestId,
+        mojaloopError as unknown as fspiopAPI.Schemas.ErrorInformationObject,
+        toParticipantId
+      )
+      // throw error to stop state machine
+      throw error
+    }
+
+    // todo: implement inbound PUT /consents/{ID}/error handler
+    //       and listen subscribe for switch errors so that they can be forwarded
+    //       back to the dfsp
+    /*
+    const channel = DFSPOTPValidateModel.notificationChannel(
+      consentRequestsRequestId
+    )
+    // eslint-disable-next-line no-async-promise-executor
+    return new Promise(async (resolve, reject) => {
+      // let subId = 0
+      try {
+        // this handler will be executed when PUT /consents/{ID}/error @ Inbound server
+        // this case should happen when something went wrong at the switch
+        // but we need to also report a switch failure back to the PISP.
+        subId = this.pubSub.subscribe(channel, async (channel: string, message: Message, sid: number) => {
+          // first unsubscribe
+          this.pubSub.unsubscribe(channel, sid)
+
+          this.data.consentsStatus = { ...message as unknown as tpAPI.Schemas.ErrorInformation }
+          resolve()
+        })
+
+        this.logger.push({ res }).info('ThirdpartyRequests.postConsents request sent to peer')
+
+        resolve()
+      } catch (error) {
+      }
+    })
+    */
+
+  }
+
+  reformatError (err: Error): Errors.MojaloopApiErrorObject {
```

Comment:
Should this be a util function implemented elsewhere? Is there any reason for it to be in this model implementation?

Reply:
Yes, it's used in a lot of places so it can be abstracted out. I've just put this here since moving it to a shared util folder would mean touching a lot of other files.